tgui/packages/tgui/interfaces/LightController.tsx

```diff
@@ -0,0 +1,319 @@
+import { useBackend, useLocalState } from '../backend';
+import { round } from '../../common/math';
+import { classes } from '../../common/react';
+import { Box, Button, Knob, Section, Slider, Stack, Tabs } from '../components';
+import { Window } from '../layouts';
+
+const NORTH = 1;
+const SOUTH = 2;
+const EAST = 4;
+const WEST = 8;
+
+type LightDetails = {
+  name: string;
+  color: string;
+  power: number;
+  range: number;
+  angle: number;
+};
+
+type TemplateID = string;
+
+type LightTemplate = {
+  light_info: LightDetails;
+  description: string;
+  id: TemplateID;
+};
+
+
+interface CategoryList {
+  [key: string]: TemplateID[];
+}
+
+type Data = {
+  on: boolean;
+  direction: number;
+  light_info: LightDetails;
+  templates: LightTemplate[];
+  default_id: string;
+  default_category: string;
+  category_ids: CategoryList;
+};
+
+export const LightController = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { light_info, templates, default_id, default_category, category_ids } = data;
+  const [currentTemplate, setCurrentTemplate] = useLocalState<string>(
+    context,
+    'currentTemplate',
+    default_id
+  );
+  const [currentCategory, setCurrentCategory] = useLocalState<string>(
+    context,
+    'currentCategory',
+    default_category
+  );
+
+
+  return (
+    <Window
+      title={light_info.name + ": Lighting"}
+      width={600}
+      height={400}>
+      <Window.Content scrollable>
+        <Stack fill>
+          <Stack.Item>
+            <Section fitted fill scrollable width="170px">
+              <Tabs fluid centered>
+                {Object.keys(category_ids).map(
```

Comment:
this should be made into a const earlier too just to be safe. something like
```ts
const categories = category_ids ? Object.keys(category_ids) : [];
```

Reply:
o7